stdlib/socket/src/main/java/org/ballerinalang/stdlib/socket/endpoint/tcp/client/InitEndpoint.java

```diff
@@ -0,0 +1,140 @@
+ * under the License.
+ */
+
+package org.ballerinalang.stdlib.socket.endpoint.tcp.client;
+
+import org.ballerinalang.bre.Context;
+import org.ballerinalang.bre.bvm.BlockingNativeCallableUnit;
+import org.ballerinalang.connector.api.BLangConnectorSPIUtil;
+import org.ballerinalang.connector.api.Resource;
+import org.ballerinalang.connector.api.Service;
+import org.ballerinalang.connector.api.Struct;
+import org.ballerinalang.connector.api.Value;
+import org.ballerinalang.model.types.TypeKind;
+import org.ballerinalang.model.values.BMap;
+import org.ballerinalang.model.values.BValue;
+import org.ballerinalang.natives.annotations.BallerinaFunction;
+import org.ballerinalang.natives.annotations.Receiver;
+import org.ballerinalang.stdlib.socket.tcp.SocketService;
+import org.ballerinalang.stdlib.socket.tcp.SocketUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.SocketException;
+import java.nio.channels.SocketChannel;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.ballerinalang.stdlib.socket.SocketConstants.CLIENT_CONFIG;
+import static org.ballerinalang.stdlib.socket.SocketConstants.CLIENT_SERVICE_CONFIG;
+import static org.ballerinalang.stdlib.socket.SocketConstants.IS_CLIENT;
+import static org.ballerinalang.stdlib.socket.SocketConstants.RESOURCE_ON_CLOSE;
+import static org.ballerinalang.stdlib.socket.SocketConstants.RESOURCE_ON_CONNECT;
+import static org.ballerinalang.stdlib.socket.SocketConstants.RESOURCE_ON_ERROR;
+import static org.ballerinalang.stdlib.socket.SocketConstants.RESOURCE_ON_READ_READY;
+import static org.ballerinalang.stdlib.socket.SocketConstants.SOCKET_KEY;
+import static org.ballerinalang.stdlib.socket.SocketConstants.SOCKET_PACKAGE;
+import static org.ballerinalang.stdlib.socket.SocketConstants.SOCKET_SERVICE;
+
+/**
+ * Initialize the client socket endpoint.
+ *
+ * @since 0.985.0
+ */
+@BallerinaFunction(
+        orgName = "ballerina",
+        packageName = "socket",
+        functionName = "initEndpoint",
+        receiver = @Receiver(type = TypeKind.OBJECT, structType = "Client", structPackage = SOCKET_PACKAGE),
+        isPublic = true
+)
+public class InitEndpoint extends BlockingNativeCallableUnit {
+    private static final Logger log = LoggerFactory.getLogger(InitEndpoint.class);
+
+    @Override
+    public void execute(Context context) {
+        try {
+            Struct clientEndpoint = BLangConnectorSPIUtil.getConnectorEndpointStruct(context);
+            Struct clientEndpointConfig = clientEndpoint.getStructField(CLIENT_CONFIG);
+            Value clientServiceType = clientEndpointConfig.getTypeField(CLIENT_SERVICE_CONFIG);
+            Service service = null;
+            if (clientServiceType != null) {
+                service = BLangConnectorSPIUtil.getServiceFromType(context.getProgramFile(), clientServiceType);
+//                if (!"ballerina/socket:Client".equals(service.getEndpointName())) {
```

Comment:
Do we need to keep these commented lines?

Reply:
Callback service not working yet due to changes in core side. These changes are pending.